### DISCUSSION TITLE
Add docstrings for public methods

### DISCUSSION THREAD

**Discussion by falkoschindler** [zauberzeug/rosys#29] [Ideas / Feature Requests]
Currently most public methods are undocumented. We should add docstrings including

- a brief description of what the method is doing,
- a description of each parameter including possible values, default value and unit (if any).

**Comment by AndyPermaRobotics** [comment]
I recently discovered this documentation tool:
https://pycco-docs.github.io/pycco/
It's lightweight and displays the documentation next to the code.

I am using excalidraw files to add drawings to my repository. 
It's JSON, so you can store it together with the source code in your repository and you can easily find and replace names if you want rename a variable.

I would love to see some drawings that explain the functionality of some code parts.
One picture says more that 1000 lines of code.
